gcp/live/dev/k8s/stackdriver-exclusions/kubelet-stats-summary-200/terraform.tfvars

```diff
@@ -0,0 +1,24 @@
+# ↓ Module metadata
+terragrunt = {
+  terraform {
+    source = "/project/modules//gcp-stackdriver-exclusion"
+  }
+
+  dependencies {
+    paths = [
+      "../../cluster",
+    ]
+  }
+
+  include = {
+    path = "${find_in_parent_folders()}"
+  }
+}
+
+# ↓ Module configuration (empty means all default)
+exclusion_name = "kubelet-stats-summary-200"
+exclusion_description = "kubelet /stats/summary requests that return status code 200"
+# TODO: Better (multi-line) string formatting
+# TODO: How to get project name into logName? Use ":/logs/kubelet" instead of exact match? Old version: 'logName=projects/gpii-gcp-dev-tyler/logs/kubelet'
+# NOTE: Trailing space on ' 200 ' is important!
+exclusion_filter = "resource.type=gce_instance AND logName:/logs/kubelet AND jsonPayload.MESSAGE:GET /stats/summary AND jsonPayload.MESSAGE:\" 200 \""
```

Comment:
Is it possible to use arrays here? Something along the lines of:
```
exclusion_filter = [
    "resource.type=gce_instance",
    "logName:/logs/kubelet",
    "jsonPayload.MESSAGE:GET /stats/summary",
    "jsonPayload.MESSAGE:\" 200 \""
]
```

And then later in `main.tf`:
```
module "gcp_stackdriver_exclusion" {
  source                = "/exekube-modules/gcp-stackdriver-exclusion"
  project_id            = "${var.project_id}"
  serviceaccount_key    = "${var.serviceaccount_key}"
  exclusion_name        = "${var.exclusion_name}"
  exclusion_description = "${var.exclusion_description}"
  exclusion_filter      = "${join(",", "${var.exclusion_filter}")}"
}
```

Reply:
The TF syntax would be `"${join(" AND ", var.exclusion_filter)}"` and you can't "just" define variables like this in TF.

Possibly you would be able to use local values (`locals`) for this.

Reply:
Thanks for the suggestions, @seg10 and @stepanstipl. I like the idea of using lists and joining with `" "` for flexibility (although the behavior in Sergio's example matches the Google UI).

Unfortunately, I'm having trouble getting this to work now that I've moved name/filter combos into a map. Apparently [`element()`](https://www.terraform.io/docs/configuration/interpolation.html#element-list-index-) and [`lookup()`](https://www.terraform.io/docs/configuration/interpolation.html#lookup-map-key-default-) don't support nested data structures.

It's always possible I'm doing it wrong. Here's a version I expected to work:

```
exclusions = {
  "calico-typha-health-summary-ok" = ["resource.type=container AND logName:/logs/calico-typha AND textPayload:\"Overall health summary=&health.HealthReport{Live:true, Ready:true}\""]
  "kubelet-healthz-200" = ["resource.type=gce_instance AND logName:/logs/kubelet AND jsonPayload.MESSAGE:GET /healthz AND jsonPayload.MESSAGE:\" 200 \""]
  "kubelet-stats-summary-200" = ["resource.type=gce_instance AND logName:/logs/kubelet AND jsonPayload.MESSAGE:GET /stats/summary AND jsonPayload.MESSAGE:\" 200 \""]
}
```

```
resource "google_logging_project_exclusion" "my-exclusion" {
  count  = "${length(var.exclusions)}"
  name   = "${element(keys(var.exclusions), count.index)}"
  filter = "${join(" ", element(values(var.exclusions), count.index))}"
}
```

```
* module.gcp_stackdriver_exclusion.google_logging_project_exclusion.my-exclusion[0]: At column 3, line 1: join: argument 1 should be type list, got type string in:

${join(" ", element(values(var.exclusions), count.index))}
```

Do you see anything I'm missing? Is there a different way to set up this data structure that might work better?

If not, I can live with goofy long lines in this one file. :)